middlewares/metrics.go

```diff
@@ -1,12 +1,35 @@
 package middlewares
 
 import (
+	"fmt"
 	"net/http"
+	"os"
 	"time"
 
+	"github.com/DataDog/datadog-go/statsd"
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+var statsdClient *statsd.Client
+
+func initStatsdClient() {
+	statsdHost := os.Getenv("STATSD_HOST")
+	statsdPort := os.Getenv("STATSD_PORT")
+	applicationNamespace := os.Getenv("STATSD_APP_NAME")
+
+	var err error
+	statsdClient, err = statsd.New(statsdHost + ":" + statsdPort)
+	if err != nil {
+		fmt.Println("** We were unable to get a statsd client.")
```

Comment:
I typically do not like to have err checks like this in a library. It is better to return the error to the caller and let them decide what do do. That way they can do logging properly rather than just printing to stdout, fail fast, fallback, turn off, whatever.